### PR TITLE
4.2.4 + Qt6

### PR DESCRIPTION
**Pull request by guihkx**
I just want to test if this fixes https://github.com/martinrotter/rssguard/issues/565, once and for all.

My current setup:

- Arch Linux
- GNOME 42.4
- RSS Guard 4.2.3 (built with Qt 5.15.4)

Currently, there doesn't seem to leak file descriptors as badly as before (presumably caused by just calling `QSystemTrayIcon::isSystemTrayAvailable()`), so that's good.

However, there is this other horrible leak triggered by just hovering links in the WebView (caused by `QWebEngineView::setHtml()`?):

https://user-images.githubusercontent.com/626206/187040448-aa6c0841-9177-4783-b56e-db482bc92b49.mp4

The commands I used in the video:

- `flatpak run com.github.rssguard -w` (to launch RSS Guard with "basic" WebView support)
- `watch -n.1 'bash -c "echo fd count: ±$(ls -l /proc/$(pidof rssguard)/fd | wc -l)"'` (to monitor fd count)

However, I've been experimenting with [Arch Linux's official build](https://archlinux.org/packages/community/x86_64/rssguard/) of RSS Guard, which only recently switched to building with Qt 6. From my testing, the same fd leak above *does not* happen in the Qt 6 build, so I want to see if this hopefully fixes the Flatpak package as well!

Closes #31

### CODE REVIEW

**Comment by guihkx** [comment]
```
-- Could NOT find Qt6Core5Compat (missing: Qt6Core5Compat_DIR)
CMake Error at CMakeLists.txt:148 (find_package):
  Found package configuration file:
    /usr/lib/x86_64-linux-gnu/cmake/Qt6/Qt6Config.cmake
  but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
  FOUND.  Reason given by package:
  Failed to find Qt component "Core5Compat".
  Expected Config file at
  "/usr/lib/x86_64-linux-gnu/cmake/Qt6Core5Compat/Qt6Core5CompatConfig.cmake"
  does NOT exist
```

Will have to investigate this one...

**Comment by guihkx** [comment]
Okay, so now it at least builds fine, but it fails to validate the dimensions of the [RSS Guard icon](https://github.com/martinrotter/rssguard/blob/master/resources/graphics/rssguard.png) for not being a perfect square. 🤡 

@martinrotter I'll add the icon here for now just to test this Qt 6 build, but for the next release, can you please fix this *\~REALLY IMPORTANT\~* icon validation issue? Thanks!

**Comment by guihkx** [comment]
Yup, it definitely fixes it:

https://user-images.githubusercontent.com/626206/187049799-1d24c102-1f7d-49a4-beb6-417c5bf55b93.mp4

Nevertheless, I'll be using this Qt 6 build for a while just to make sure there are no obvious bugs.

**Comment by guihkx** [comment]
Okay, so after leaving RSS Guard running for about 3 days straight, I can confirm that the file descriptor leak issue is gone for good!

The switch to Qt 6 causes only two minimal differences, but they should only affect GNOME users with a very specific setup.

The first issue is simply the placeholder text in input fields being hard to read (reported to upstream [here](https://github.com/FedoraQt/adwaita-qt/issues/176)), but it only affects people using QGnomePlatform/adwaita-qt, which are components used to better integrate Qt apps running on GNOME (theming-wise).

The second issue is related to system tray notifications. On the Qt 5 build, notifications were sent directly to GNOME's "notification center", also thanks to QGnomePlatform, but now, they are displayed as a balloon like this (which I think it's how it is on KDE?):

Screenshot from 2022-08-30 05-12-04

For me, this is not a big deal and it's actually better to read than GNOME's notification center. Nevertheless, [this has been already fixed](https://github.com/FedoraQt/QGnomePlatform/issues/89), but there's no release with the fix just yet.

To sum up, if you are okay with the changes, I think this PR could be postponed to the next RSS Guard release, so I can drop the patches and make the PR cleaner.

**Comment by martinrotter** [comment]
In fact, I believe that this could/should be merged right now. Feel free to rearrange the PR to make it cleaner and then let me know and I will merge.

**Comment by guihkx** [comment]
Awesome! Well, the PR is pretty much ready at this point. Just keep in mind that if you want to merge this right now, on the next RSS Guard release you'll have to modify the manifest a bit and remove the left over files, because the two fixes that I'm using here have already been upstreamed by you [here](https://github.com/martinrotter/rssguard/commit/831029d9a511ebc874372674da3c6b517cb2df10) and [here](https://github.com/martinrotter/rssguard/commit/605c1b0afc2a13cd25e01883bd8380890a6cad2d).

So for the next release you'd just delete lines 56 through 65 and then delete these two files: `qt6-build-fix.patch` and `rssguard-perfect-square.png`.

**Comment by guihkx** [comment]
@martinrotter Please merge this one instead of #31 (which I also included here). Thank you!
